tests/test_otdd.py

```diff
@@ -0,0 +1,47 @@
+import os
+
+from modsim2.data.loader import DMPair
+from modsim2.utils.config import load_configs
+
+# project root = arc-model-similarites-phase-2/
+PROJECT_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+
+# Get metric config
+METRICS_CONFIG_PATH = os.path.join(PROJECT_ROOT, "tests", "testconfig", "metrics.yaml")
+CONFIGS = load_configs(METRICS_CONFIG_PATH)
+METRIC_CONFIG = CONFIGS["metric_config"]
+METRIC_CONFIG = {k: v for k, v in METRIC_CONFIG.items() if v["class"] == "otdd"}
+
+
+# This test checks that the distance between a dataset and itself is the expect value
+# When exact calculations are used then this will be zero
+# Approximate methods may be non-zero and these are checked against the known value
+# for the seed
+def test_cifar_otdd_same():
+    dmpair = DMPair(metric_config=METRIC_CONFIG, seed=42)
+    similarity_dict = dmpair.compute_similarity(only_train=False, metric_seed=42)
+    for k in similarity_dict:
```

Comment:
The suggestion looks neater, but not sure it is more efficient? You would still have to loop over k, i.e. 

```
for k in METRIC_CONFIG:
    expected_trainval = METRIC_CONFIG[k]["expected_results"]["same_result"]
    expected_trainonly = METRIC_CONFIG[k]["expected_results"]["same_result_only_train"]
    assert dmpair.compute_similarity(only_train=False, metric_seed=42)[k] == expected_trainval
    assert dmpair.compute_similarity(only_train=True, metric_seed=42)[k] == expected_trainonly
```

And then compute_similarity will be calculated for value of k. Could create one `dmpair` for each `k` so that it's not recalculating compute_similarity for all k each time, but that would mean the datasets are created for each `k` which isn't good either. 

I could reduce it to one `for` loop by naming the `similarity_dict` variables differently.